homeassistant/components/pure_energie/config_flow.py

```diff
@@ -0,0 +1,119 @@
+
+    VERSION = 1
+    discovered_host: str
+    discovered_device: Device
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            try:
+                device = await self._async_get_device(user_input[CONF_HOST])
+            except GridNetConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(device.n2g_id)
+                self._abort_if_unique_id_configured(
+                    updates={CONF_HOST: user_input[CONF_HOST]}
+                )
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                    },
+                )
+        else:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
+                    ): str,
+                    vol.Required(CONF_HOST): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        self.discovered_host = discovery_info.host
+        try:
+            self.discovered_device = await self._async_get_device(discovery_info.host)
+        except GridNetConnectionError:
+            return self.async_abort(reason="cannot_connect")
+
+        await self.async_set_unique_id(self.discovered_device.n2g_id)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.host})
+
+        self.context.update(
+            {
+                "title_placeholders": {
+                    CONF_NAME: "Pure Energie Meter",
+                    CONF_HOST: self.discovered_host,
+                    "model": self.discovered_device.model,
+                },
+            }
+        )
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by zeroconf."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=user_input[CONF_NAME],
+                data={
+                    CONF_HOST: self.discovered_host,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
```

Comment:
Same as above, we don't have to ask this.

Reply:
Removed